### PR TITLE
[WIP] Add the message size in the trace

### PR DESCRIPTION
**Pull request by Ezibenroc**
This is a tentative to add the message size in the trace. You should not merge it yet, it is broken at this line of `aky_converter.c`:
```c
poti_user_PushState (specialPushState, timestamp, mpi_process, "STATE", value, nb_extra, extra);
```

The function `poti_user_PushState` is a variadic function, I have to pass the additional fields of the event. The problem is, I don't know how to pass a variable number of arguments.

Is there a reason why this function doesn't simply take an array of `char*`?

### CODE REVIEW

**Comment by schnorr** [comment]
Hi @Ezibenroc, the idea of `poti_user_PushState` is to enforce some obligatory parameters. Poti guarantees then that the enforced parameters will be handled consistently with the header, which might not be the case for extra user arguments. The idea is to have some sort of protection against poti users that are unaware of the file format and that use the API to guide themselves during trace modeling.

File: src/aky/aky_converter.c [hunk 1]
```diff
@@ -428,10 +431,13 @@ int main(int argc, char **argv)
           snprintf(extra[i_extra], 21, "%"PRIu64, event.v_uint64[i_array]);
         }
         if(nb_extra) {
-          poti_user_PushState (specialPushState, timestamp, mpi_process, "STATE", value, nb_extra, extra);
+          poti_user_PushState_buffer (specialPushState, timestamp, mpi_process, "STATE", value, nb_extra, extra);
```
Comment (schnorr) [inline src/aky/aky_converter.c]: Is there any reason for the `_buffer` variant? Such function remains undefined in the current PR.

Comment (Ezibenroc) [reply src/aky/aky_converter.c]: This variant is defined by this commit in my fork of Poti: https://github.com/Ezibenroc/poti/commit/d606bbd50656183328a699a6a432611233c8137e

I wanted to pass an arbitrary number of extra arguments, which was not possible with the variadic version of the function (or I don't know how).

**Comment by Ezibenroc** [comment]
> Hi @Ezibenroc, the idea of poti_user_PushState is to enforce some obligatory parameters. Poti guarantees then that the enforced parameters will be handled consistently with the header, which might not be the case for extra user arguments. The idea is to have some sort of protection against poti users that are unaware of the file format and that use the API to guide themselves during trace modeling.

I am not sure to understand, you do not want to use extra arguments here? How would you passe the message size?

**Comment by schnorr** [comment]
That's okay to use extra arguments. I just wrote that the obligatory parameters should be explicit in the function call. And by obligatory I mean that minimum set of parameters necessary for a correct Paje trace file.

**Pull request merged**